helm/prefect-server/values.yaml

```diff
@@ -0,0 +1,273 @@
+
+  # postgresql configuration below here is only used if using
+  # the subchart
+
+  # persistence enables a PVC that stores the database between
+  # deployments. If making changes to the database deployment, this
+  # PVC will need to be deleted for database changes to take effect.
+  # This is especially notable when the authentication password
+  # changes on redeploys.
+  # This is disabled by default because we do not recommend using
+  # the subchart deployment for production deployments.
+  persistence:
+    enabled: false
+    size: 8Gi
+
+  initdbUser: postgres
+
+  initdbScripts:
+    create_pgcrypto.sql: |
+      -- create pgcrypto extension, required for Hasura UUID
+      CREATE EXTENSION IF NOT EXISTS pgcrypto;
+      CREATE EXTENSION IF NOT EXISTS "pg_trgm";
+      SET TIME ZONE 'UTC';
+
+# prefectConfig allows override of prefect_server/config.toml values
+# Values placed here will result in environment variable definitions
+# starting with "PREFECT_SERVER__" and will be passed to TODO
+prefectConfig:
+  env: "local"
+  debug: "false"
+  queued_runs_returned_limit: "25"
+  hasura:
+    execute_retry_seconds: "10"
+  logging:
+    level: "DEBUG"
+    format: "[%(asctime)s] %(levelname)s - %(name)s | %(message)s"
+  services:
+    scheduler:
+      scheduler_loop_seconds: "300"
+    lazarus:
+      resurrection_attempt_limit: "3"
+  telemetry:
+    enabled: "true"
+
+
+# hasura configures the Prefect hasura deployment and service
+# which creates a graphql api from the postgres database
+hasura:
+  port: 3000
+
+  # hasura.image configures the docker image used for hasura
+  # and is the only image in the chart that is not hosted by
+  # Prefect
+  image:
+    repository: hasura/graphql-engine
+    version: v1.3.0
+    pullPolicy: null
+    pullSecrets: []
+
+  options:
+    queryPlanCacheSize: "100"
+
+  service:
+    # type defines the service type and defaults to ClusterIP
+    # because this service does not need to be exposed outside
+    # the cluster
+    type: ClusterIP
+
+  labels: {}
+  annotations: []
+  replicaCount: 1
+  podSecurityContext: {}
+  securityContext: {}
+  resources: {}
+  nodeSelector: {}
+  tolerations: []
+  affinity: {}
+
+
+# graphql configures the Prefect graphql deployment and service
+# which provides a python graphql server on starlette
+graphql:
+  port: 4201
+
+  # containerPort is currently hardcoded in the container and
+  # should not be changed (?)
+  containerPort: 3000
+
+  image:
+    repository: prefecthq/server
+    version: null
+    pullPolicy: null
+    pullSecrets: []
+
+  service:
+    type: ClusterIP
+
+  labels: {}
+  annotations: null
+  replicaCount: 1
+  podSecurityContext: {}
+  securityContext: {}
+  resources: {}
+  nodeSelector: {}
+  tolerations: []
+  affinity: {}
+  init:
+    resources: {}
+
+# apollo configures the Prefect apollo deployment and service
+# which provides a unified graphql schema for users and the UI
+# to interact with
+apollo:
+  port: 4200
```

Comment:
Typically helm charts will use fixed internal ports (e.g. the apollo pods would all listen on a hardcoded port inside their pod), and only external-facing ports are configurable. I didn't trace through the helm chart deeply, but on first glance it looks like you're exposing both (the port here for setting the pod-internal ports, and the one in the service for external-facing ports).

I suggest we only expose the external `service` ports, and hardcode the internal ones - the user should have no need to configure these.